src/crm.ts

```diff
@@ -0,0 +1,113 @@
+import { ActionMessageV2, EventMessageV2 } from "@proca/queue";
+export type handleResult = {
+  pause?: 0; // should it wait delay seconds before reading the next message?
+  processed: boolean; // if the message has been processed or skipped
+};
+
+export { ActionMessageV2 as ActionMessage };
+export { EventMessageV2 as EventMessage };
+
+type Obj = { [key: string]: any };
+
+export type ProcaCampaign = { [key: string]: any }; // TODO export from proca/queue
+
+export type Contact = {
+  area: string;
+  contactRef: string;
+  country: string;
+  dupeRank: number;
+  email: string;
+  firstName: string;
+  lastName: string;
+  postcode: string;
+};
+
+export type Privacy = {
+  //  emailStatus: null,
+  //  emailStatusChanged: null,
+  givenAt: string;
+  optIn: boolean;
+  withConsent: boolean;
+};
+
+export type Action = {
+  actionType: string;
+  createdAt: string;
+  customFields: object;
+  testing: boolean;
+};
+
+interface CRMInterface {
+  handleActionContact: (
+    message: ActionMessageV2
+  ) => Promise<handleResult | void>;
+  // TODO: proca/queue exports Action and ContactType, probably source and ap too?
+  handleContact?: (message: ActionMessageV2) => Promise<handleResult|void>,
+  handleEvent?: (message: EventMessageV2) => Promise<handleResult | void>; //WIP
+  handleEmailStatusChange?: (
+    message: EventMessageV2
+  ) => Promise<handleResult | void>; //WIP
+  campaign: (campaign: ProcaCampaign) => Promise<any>; // get the extra data from the campaign
+  fetchCampaign?: (campaign: ProcaCampaign) => Promise<any>; // fetch the campaign extra data and store it locally
+}
+
+export enum CRMType {
+  ActionContact, // Process all actions, primary and secondary, optin and optout
+  Contact, // Process only the primary actions (eg petition signature, mtt), not the secondary ones (eg share actions)
+  OptIn, // Process only the primary actions of the contacts that opted in
+  //  DoubleOptIn, @marcin, can we easily do that? it'd need to memstore temporarily contacts until the doubleoptin arrives, right?
```

Comment:
in case we want to handle email opt in after the action was accepted, we would indeed need to hold the actions somewhere.
But if they do not store actions waiting for an email opt in, how can they even know how many are there, or how would they send a reminder or something like this? 
If they want to hold the actions in proca until clicked, then they can use email confirm feature and that will release the actions only after member clicks (and selects email opt in or not) - it will arrive all at once - no need to store the action.

Reply:
it seems most org want the double opt-in but not the confirm (that seems to be linked to "high cost/bar" actions.

you're probably right, no need store for the v1, probably would make more sense when handling reminders too.

Reply:
In the event you have the `contact` with same contact data as on action.
So perhaps for CRMs that just want to add contacts (like Emarsys) we do not need to store anything to create that user.
The data is missing source action.. .but when member double opt ins, they already have few actions.. let me thing what data could be useful here.
Maybe i can just add the action that contained this pii...

Reply:
Addressing this issue with a simple solution here: https://github.com/fixthestatusquo/proca-server/issues/191